src/Storages/MergeTree/IMergeTreeDataPart.h

```diff
@@ -419,6 +429,10 @@ class IMergeTreeDataPart : public std::enable_shared_from_this<IMergeTreeDataPar
 
     std::map<String, std::shared_ptr<IMergeTreeDataPart>> projection_parts;
 
+    std::map<String, std::shared_ptr<IMergeTreeDataPart>> virtual_projection_parts;
+
+    bool is_virtual = false;
```

Comment:
Can we add more comments here? What is `projection_parts` and `virtual_projection_parts`, what is the difference. What `is_virtual ` means for part and when such a part can be created/used for.